app/src/main/java/nl/eduvpn/app/fragment/ProviderSelectionFragment.java

```diff
@@ -118,29 +130,35 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         View view = inflater.inflate(R.layout.fragment_provider_selection, container, false);
         _unbinder = ButterKnife.bind(this, view);
         EduVPNApplication.get(view.getContext()).component().inject(this);
-        _providersList.setHasFixedSize(true);
-        _providersList.setLayoutManager(new LinearLayoutManager(view.getContext(), LinearLayoutManager.VERTICAL, false));
-        _providersList.setAdapter(new ProviderAdapter(_configurationService, _authorizationType));
-        ItemClickSupport.addTo(_providersList).setOnItemClickListener(new ItemClickSupport.OnItemClickListener() {
+        if (_authorizationType == AuthorizationType.LOCAL) {
+            _header.setText(R.string.select_your_institution_title);
+            _description.setVisibility(View.GONE);
+        } else {
+            _header.setText(R.string.select_your_country_title);
+            _description.setVisibility(View.VISIBLE);
+        }
+        _providerList.setHasFixedSize(true);
+        _providerList.setLayoutManager(new LinearLayoutManager(view.getContext(), LinearLayoutManager.VERTICAL, false));
+        final ProviderAdapter adapter = new ProviderAdapter(_configurationService, _authorizationType);
+        _providerList.setAdapter(adapter);
+        ItemClickSupport.addTo(_providerList).setOnItemClickListener(new ItemClickSupport.OnItemClickListener() {
             @Override
             public void onItemClicked(RecyclerView recyclerView, int position, View v) {
                 Instance instance = ((ProviderAdapter)recyclerView.getAdapter()).getItem(position);
                 if (instance == null) {
-                    if (getActivity() != null) {
-                        MainActivity activity = (MainActivity)getActivity();
-                        Fragment customProviderFragment = new CustomProviderFragment();
-                        Bundle fragmentParameters = new Bundle();
-                        fragmentParameters.putInt(CustomProviderFragment.EXTRA_AUTHORIZATION_TYPE, _authorizationType);
-                        customProviderFragment.setArguments(fragmentParameters);
-                        activity.openFragment(customProviderFragment, true);
+                    // Should never happen
+                    View mainView = getView();
+                    if (mainView != null) {
+                        Snackbar.make(mainView, R.string.error_selected_instance_not_found, Snackbar.LENGTH_LONG).show();
                     }
+                    Log.e(TAG, "Instance not found for position: " + String.valueOf(position));
```

Comment:
No Timber?

Reply:
Since it is an open-source app, it is not really necessary to hide the logs.
Of course I don't log any sensitive info.